plenty_api/api.py

```diff
@@ -518,43 +524,85 @@ def __repeat_get_request_for_all_unpaginated_records(self,
         page_info = utils.sniff_response_format(response=response)
         entries = response[page_info['data']]
 
-        while True:
-            logging.debug(f"fetching page {query['page'] + 1}")
-            query.update({'page': query['page'] + 1})
-            response = self.__plenty_api_request(method='get',
-                                                 domain=domain,
-                                                 path=path,
-                                                 query=query)
-            if not response:
-                return None
+        if len(response[page_info['data']]) == query['itemsPerPage']:
+            while True:
+                logging.debug(f"fetching page {query['page'] + 1}")
+                query.update({'page': query['page'] + 1})
+                response = self.__plenty_api_request(method='get',
+                                                    domain=domain,
+                                                    path=path,
+                                                    query=query)
+                if not response:
+                    return None
 
-            if isinstance(response, dict) and 'error' in response.keys():
-                logging.error(f"subsequent {domain} API requests failed.")
-                return response
+                if isinstance(response, dict) and 'error' in response.keys():
+                    logging.error(f"subsequent {domain} API requests failed.")
+                    return response
 
-            entries += response[page_info['data']]
+                entries += response[page_info['data']]
 
-            if len(response[page_info['data']]) < query['itemsPerPage']:
-                break
+                if len(response[page_info['data']]) < query['itemsPerPage']:
+                    break
 
         return entries
 
 
+    def __get_request_dump_file(self,
```

Comment:
After thinking about it a bit more, I believe this should be public method.
The `plenty_api_get_bi_raw_files` method, returns a bunch of data structures describing files. Currently, the user can only choose to either download them all or download none. I think when we present the user with data, he should be able to work with that data and then decide which files to download.
So, my suggestion would be to rename the method to: `plenty_api_dump_bi_raw_file` and that it should accept either a single download_filepath or a list of download_filepaths. That way the user can choose specific files or download all of them.

Reply:
My intention was to filter the files via the refine options, but after playing around a bit and thinking about it, I totally agree with you. It gives the whole thing more flexibility.